elastica/_rotations.py

```diff
@@ -146,9 +146,13 @@ def _inv_rotate(director_collection):
             )
         )
 
-        # TODO HARDCODED bugfix has to be changed. Remove 1e-14 tolerance
-        theta = arccos(0.5 * trace - 0.5 - 1e-10)
+        # Clip the trace to between -3 and 3.
+        # Any deviation beyond this is numerical error
+        trace = min(trace, 3.0)
+        trace = max(trace, -1.0)
+        theta = arccos(0.5 * trace - 0.5)
 
+        # TODO HARDCODED bugfix has to be changed. Remove 1e-14 tolerance
```

Comment:
Should fix this together with other tolerance cases in a future version. This arbitrary tolerancing may cause some serious issues especially in the anisotropic plane friction